### ISSUE TITLE
DialsScaler.py should not conflate the concept of logical WAVELENGTH with physical X-ray wavelength

### ISSUE THREAD

**Issue by benjaminhwilliams** [xia2/xia2#560]
TL;DR, in the `DialsScaler`:
- [ ] Remove the assertion that the number of `WAVELENGTH`s equals the number of unique physical wavelengths.
- [ ] Don't split by physical wavelength.  Rather, `dials.split <PROJECT>_<CRYSTAL>_scaled_unmerged.{expt,refl}` and then `dials.combine` the split experiment lists and reflection tables into one pair of files per `WAVELENGTH`.

In xia2, an integration and data reduction job is organised with a logical hierarchy:  one `PROJECT`, which has one or more `CRYSTAL`s, each of which may have one or more `WAVELENGTH`s, each of which in turn may have one or more `SWEEP`s.  All the data that are grouped as a single `CRYSTAL` are scaled together.  Then all the data that are grouped as a single `WAVELENGTH` are merged together.  Most of the time, you would only have multiple `WAVELENGTH`s if you have collected data with multiple physical X-ray wavelengths, such as in a MAD experiment.  However, xia2 is designed for the `WAVELENGTH` categorisation to be co-opted in cases where you want to merge data from groups of sweeps, group-by-group, even though they may all have been collected with the same physical wavelength.  In this case, `WAVELENGTH` becomes a misnomer,  it is really just a categorisation meaning 'group of `SWEEP`s to be merged together'.

This mode of use is not well documented or adequately tested, so it is unsurprising that it is actually broken in the default all-DIALS pipeline. `DialsScaler` assumes a one-to-one equivalence between the `WAVELENGTH` labels and the physical wavelengths at which the data were measured.  For example, here it asserts that the number of unique physical wavelengths matches the number of `WAVELENGTH` labels:
https://github.com/xia2/xia2/blob/b4d7ac14179b09815a00566be02baf1837b1e8a5/src/xia2/Modules/Scaler/DialsScaler.py#L675-L684
And here it splits the `<PROJECT>_<CRYSTAL>_scaled_unmerged.{expt,refl}` by physical wavelength (using `dials.split_experiments by_wavelength=True`), before passing each resulting pair of `.expt` & `.refl` to `dials.merge`.  It ought to be grouping by `WAVELENGTH`, rather than by wavelength, and those two concepts are not necessarily equivalent:
https://github.com/xia2/xia2/blob/b4d7ac14179b09815a00566be02baf1837b1e8a5/src/xia2/Modules/Scaler/DialsScaler.py#L695-L698

**Comment by graeme-winter** [comment]
Proposed test case:

Take existing single wavelength xia2 test case, write new `xinfo` file which contains the data split into two logical wavelengths. Run xia2 on these. Verify that there are two "wavelengths" when the data processing completes (currently the processing will simply fail.)

I will write this test in a moment.

**Comment by graeme-winter** [comment]
Minimal test (to be improved) in `logical-wavelength` branch
